ewitem.py

```diff
@@ -1396,6 +1398,115 @@ def gen_item_props(item):
+				response = "That's not your soul. Nice try, though."
+			else:
+				response = "You open the soul jar and hold the opening to your chest. The soul begins to crawl in, and a warmth returns to your body. Not exactly the warmth you had before, but it's too wonderful to pass up. You feel invigorated and ready to take on the world."
+				item_delete(id_item=soul.get('id_item'))
+				usermodel.has_soul = 1
+				usermodel.persist()
+		else:
+			response = "Nice try, but your mortal coil recognizes a fake soul when it sees it."
+	else:
+		response = "You don't have a soul to absorb. Hopelessness is no fun, but don't get all delusional on us now."
+	return await ewutils.send_message(cmd.client, cmd.message.channel, ewutils.formatMessage(cmd.message.author, response))
+
+async def squeeze(cmd):
+	usermodel = EwUser(member=cmd.message.author)
+	soul_inv = inventory(id_user=cmd.message.author.id, id_server=cmd.message.server.id, item_type_filter=ewcfg.it_cosmetic)
+
+	if cmd.mentions_count <= 0:
+		response = "Specify a soul you want to squeeze the life out of."
+	elif cmd.mentions_count > 1:
+		response = "One dehumanizing soul-clutch at a time, please."
+	else:
+		target = cmd.mentions[0]
+		targetmodel = EwUser(member = target)
+		playermodel = EwPlayer(id_user=targetmodel.id_user)
+		receivingreport = "" #the receiver of the squeeze gets this in their channel
+
+		squeezetext = re.sub("<.+>", "", cmd.message.content[(len(cmd.tokens[0])):]).strip()
+		if len(squeezetext) > 500:
+			haunt_message_content = squeezetext[:-500]
+
+
+		poi = None
+		target_item = None
+		for soul in soul_inv:
+			soul_item = EwItem(id_item=soul.get('id_item'))
+			if soul_item.item_props.get('user_id') == targetmodel.id_user:
+				target_item = soul
+
+		if targetmodel.has_soul == 1:
+			response = "They look pretty soulful right now. You can't do anything to them."
+		elif target_item == None:
+			response = "You don't have their soul."
+		elif (int(time.time()) - usermodel.time_lasthaunt) < ewcfg.cd_squeeze:
+			timeleft = ewcfg.cd_squeeze - (int(time.time()) - usermodel.time_lasthaunt)
+			response = "It's still all rubbery and deflated from the last time you squeezed it. Give it {} seconds.".format(timeleft)
+		else:
+			if squeezetext != "":
+				receivingreport = "A voice in your head screams: \"{}\"\nSuddenly, you feel searing palpitations in your chest. Dammit, {} must be fucking around with your soul.".format(squeezetext, playermodel.display_name)
+			else:
+				receivingreport = "You feel searing palpitations in your chest. Dammit, {} must be fucking with your soul.".format(playermodel.display_name)
+
+			poi = ewcfg.id_to_poi.get(targetmodel.poi)
+
+			usermodel.time_lasthaunt = int(time.time())
+			usermodel.persist()
+
+			server = ewcfg.server_list[targetmodel.id_server]
+			member_object = server.get_member(targetmodel.id_user)
+
+			penalty = (targetmodel.slimes* -0.25)
+			targetmodel.change_slimes(n=penalty, source=ewcfg.source_haunted)
```

Comment:
is it supposed to work on ghosts? cause currently it does

Reply:
it was, but it might be better if it didn't.